amxmodx/amx.cpp

```diff
@@ -4185,9 +4185,9 @@ int AMXAPI amx_ExecPerf(AMX* amx, cell* retval, int index)
         CPluginMngr::CPlugin* perf_Plug = g_plugins.findPluginFast(amx);
         if (perf_Plug && perf_Plug->isDebug())
         {
-            char perf_funcname[512];
+            char perf_funcname[sNAMEMAX + 1];
             amx_GetPublic(perf_Plug->getAMX(), index, perf_funcname);
-            if (strlen(perf_funcname) == 0)
+            if (perf_funcname[0] == '\0')
```

Comment:
Sorry one more thing here, I just realized - amx_GetPublic can return an error, in which case the string won't be initialized. So you could fix this either by:

    char perf_funcname[sNAMEMAX + 1] = {};

Or:

    int err = amx_GetPublic(perf_Plug->getAMX(), index, perf_funcname);
    if (err != AMX_ERR_NONE)
        snprintf(perf_funcname, sizeof(perf_funcname), "Unknown_ID%d", index);

Reply:
Yes, really. Thanks.

Reply:
I think now it is fixed?